app/src/main/java/org/oppia/android/app/activity/route/ActivityRouter.kt

```diff
@@ -0,0 +1,36 @@
+package org.oppia.android.app.activity.route
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import org.oppia.android.app.model.DestinationScreen
+import org.oppia.android.app.model.DestinationScreen.DestinationScreenCase
+import org.oppia.android.util.logging.ConsoleLogger
+import javax.inject.Inject
+
+/**
+ * A central router that can navigate the user to a specific activity based on a provided
+ * [DestinationScreen].
+ */
+class ActivityRouter @Inject constructor(
+  private val activity: AppCompatActivity,
+  private val destinationRoutes: Map<DestinationScreenCase, @JvmSuppressWildcards Route>,
+  private val consoleLogger: ConsoleLogger
+) {
+  /**  Opens the activity corresponding to the specified [destinationScreen]. */
+  fun routeToScreen(destinationScreen: DestinationScreen): Intent? {
```

Comment:
I have changed this function and instead of starting an Activity inside this function, I am returning Intent and passing it further in activities. I have changed the logic to make the code testable as if I directly startActivity inside this function it is difficult to test code.

Reply:
What is the specific issue that you're running into? It seems much more preferable to perform the startActivity() call here, so I think it needs to be really clear why we can't do that.

Reply:
Changed back to using startActivity() inside routeToScreen